File: .tmp_c4000/batch_04_go/3467_transform_array_by_parity.go
// LeetCode 3467 - Transform Array by Parity
// https://leetcode.com/problems/transform-array-by-parity/

func transformArray(nums []int) []int {
	for i, x := range nums {
		nums[i] = x % 2
	}
	// sort: zeros then ones
	j := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			nums[i], nums[j] = nums[j], nums[i]
			j++
		}
	}
	return nums
}

File: .tmp_rs4013/rust_check17/3467_transform_array_by_parity.rs
struct Solution;
// LeetCode 3467 - Transform Array by Parity
// https://leetcode.com/problems/transform-array-by-parity/

impl Solution {
    pub fn transform_array(mut nums: Vec<i32>) -> Vec<i32> {
        for x in nums.iter_mut() {
            *x %= 2;
        }
        let mut j = 0;
        for i in 0..nums.len() {
            if nums[i] == 0 {
                nums.swap(i, j);
                j += 1;
            }
        }
        nums
    }
}

fn main() {}
